src/Encoder.ts

```diff
@@ -200,7 +213,7 @@ export class Encoder<ContextType = undefined> {
           // uint 32
           this.writeU8(0xce);
           this.writeU32(object);
-        } else if (!this.useBigInt64) {
+        } else if (this.useInt64) {
```

Comment:
The `useBigInt64` option was being conflated for 2 things - one was encoding bigint's as int64, the other was encoding big `number`'s as `int64` rather than `float64` so I've added configuration to support both (`useInt64` and `forceBigIntToInt64`). The default config (both `false`) will have the same behaviour as before.

Reply:
Actually, on reflection, maybe the default behaviour here is different and I need to make `useInt64` true by default?

Reply:
Here's my opinion. The upstream library's decision to encode numbers above the uint32 max or below the int32 min if `useBigInt64` is false is nonsensical. Javascript numbers can safely represent integers outside of this range without using bigints, and the decision to encode these as floats when using the default encoding params (thereby confusing every other non-JS msgp decoder) is mind boggling.

I'd get rid of this condition entirely.

At this point our top priority is delivering a library with enhanced features that are easy to use and make sense. We're only in this position because the upstream library has failed us. Let's focus on correcting the behavior over trying to be compatible with it.

Reply:
I'm happy to drop compatibility with the core library if you are. My main concern is that it means we do have to maintain this library long term since it makes it less likely that the core library would accept our changes if they take on maintenance of that library again.

Side-note: should we raise all of these PRs to the core library as well as this one do you think or not worth it?

Reply:
Also, the fact there is a forceIntegerToFloat means that it's still possible to force a BigInt to a float.